src/org/traccar/BaseProtocol.java

```diff
@@ -67,4 +86,25 @@ public void sendCommand(ActiveDevice activeDevice, Command command) {
         }
     }
 
+    public void setSmsEncoder(StringProtocolEncoder smsEncoder) {
+        this.smsEncoder = smsEncoder;
+    }
+
+    @Override
+    public void sendSmsCommand(String phone, Command command) throws RecoverablePduException, UnrecoverablePduException,
+            SmppTimeoutException, SmppChannelException, InterruptedException {
+        if (Context.getSmppManager() != null) {
+            if (command.getType().equals(Command.TYPE_CUSTOM)) {
+                Context.getSmppManager().sendMessageSync(phone, command.getString(Command.KEY_DATA), true);
+            } else if (supportedSmsCommands.contains(command.getType()) && smsEncoder != null) {
+                Context.getSmppManager().sendMessageSync(phone, (String) smsEncoder.encodeCommand(command), true);
+            } else {
+                throw new RuntimeException(
+                        "Command " + command.getType() + " is not supported in protocol " + getName());
+            }
+        } else {
+            throw new RuntimeException("Smpp client is not enabled");
```

Comment:
Change "Smpp" to "SMPP" here. Camel-case naming convention applies to Java code, but log messages should follow English language rules. SMPP is an abbreviation, so should be all upper case, same as SMS.